server/service/orbit.go

```diff
@@ -130,6 +132,10 @@ func (svc *Service) EnrollOrbit(ctx context.Context, hostInfo fleet.OrbitHostInf
 
 	secret, err := svc.ds.VerifyEnrollSecret(ctx, enrollSecret)
 	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			// OK: We are either in the middle of a secret rotation or the secret is invalid ...
```

Comment:
Just in case: There's no secret rotation for the enroll secret like we do for Fleet Desktop token.

This can happen if: 1. user deletes global/team enroll secret, then 2. user deletes the host in Fleet and then 3. Orbit tries to re-enroll using old secret.